equinox_monorepo/neural_cde/neural_cde.py

```diff
@@ -0,0 +1,190 @@
+from collections.abc import Callable
+from enum import Enum
+
+import diffrax as dfx
+import equinox as eqx
+import jax
+import jax.nn as jnn
+import jax.numpy as jnp
+import jax.random as jr
+import jax.tree_util as jtu
+from jaxtyping import Array, Float, PRNGKeyArray, PyTree, Scalar
+from typing_extensions import TypeAlias
+
+# These type aliases are unnecessary, but I (packquickly) like them :)
+Linear: TypeAlias = eqx.nn.Linear
+MLP: TypeAlias = eqx.nn.MLP
+Interpolator: TypeAlias = dfx.AbstractGlobalInterpolation
+Solver: TypeAlias = dfx.AbstractSolver
+
+
+class Mode(Enum):
```

Comment:
The model only offers a few options ("terminal", "meanpool") which the user may want to switch between, so in this case I prefer enums over something like a string (better type safety, avoid misspellings, are easier to refactor, self-documenting, etc.)

Reply:
I've not used an Enum as a class attribute before, but my understanding is that this would require you to import Mode from this file alongside NCDEClassifier in order to swap to meanpool? I feel this is not as clean as being able to import just NCDEClassifier and specify all of its attributes.

Reply:
If the above is the case, but you would still prefer to use an Enum, then I think that the name should be made more specific, like `PoolingModeNCDE`, so its use is clearer when imported into other scripts

Reply:
You're right, I think enums as class attributes in general feel "unequinox." Decided to simply remove it in favor of a bool switching between meanpool and terminal, since these two approaches dominate and this is a simple model already. An alternative would be to pass a callable, which gives the user more flexibility but is too complex for this model in my opinion.